### PR TITLE
Updates for hub hackathon, mimecast, minfraud, misp, mxtoolbox_dns, n…

### PR DESCRIPTION
**Pull request by tylerslijboom-r7**
…etmiko, networktotal, newrelic, nmap, viper, virustotal_yara, vmray, vxstream_sandbox, wazuh_ossec, whois, wigle, wordpress, zendesk, zenhub

Currently, docker validator fails on netmiko and wazuh_ossec plugins.  I am currently testing those two with icon-validate again on my machine, but on the master branch.  

inspecting the code, they seem to NOT be obsolete...

### CODE REVIEW

File: minfraud/plugin.spec.yaml [hunk 1]
```diff
@@ -6,20 +6,20 @@ title: minFraud
 vendor: rapid7
 support: community
 status: []
-description: The MaxMind minFraud services help identify fraudulent online transactions
+description: Help identify fraudulent online transactions, not just e-commerce; but user account creation, and application installations
 version: 1.0.1
```
Comment (jschipp-r7) [inline minfraud/plugin.spec.yaml]: This needs to be a major version bump since titles changed.

File: mxtoolbox_dns/help.md [hunk 2]
```diff
@@ -1,11 +1,17 @@
 # Description
 
-[mxtoolbox_dns](https://www.mxtoolbox.com) is a Interface with MxToolBox DNS lookup API
```
Comment (jschipp-r7) [inline mxtoolbox_dns/help.md]: We should be keeping the links

File: misp/help.md [hunk 3]
```diff
@@ -1,17 +1,15 @@
 # Description
 
-[MISP](http://www.misp-project.org/) is an open source threat sharing platform.
-This plugin utilizes the [MISP API](https://circl.lu/doc/misp/automation/index.html) and leverages the [pymisp](https://github.com/CIRCL/PyMISP) library.
+Open source threat sharing platform.  Gather, store and then find correlations of indicators of compromise
```
Comment (jschipp-r7) [inline misp/help.md]: Keep original link, missing period at end of sentence. This long description should contain more information about the capabilities of the plugin.

File: misp/help.md [hunk 4]
```diff
@@ -1,17 +1,15 @@
...
 
-* Library of known threats
-* Global sharing platform of know threats
+Identify key features of plugin.
 
 # Requirements
 
-* MISP server
-* Username and Password
+* Example: Requires an API Key from the product
```
Comment (jschipp-r7) [inline misp/help.md]: Examples got introduced

File: misp/help.md [hunk 5]
```diff
@@ -1,17 +1,15 @@
...
 
-[MISP](http://www.misp-project.org/) is an open source threat sharing platform.
-This plugin utilizes the [MISP API](https://circl.lu/doc/misp/automation/index.html) and leverages the [pymisp](https://github.com/CIRCL/PyMISP) library.
+Open source threat sharing platform.  Gather, store and then find correlations of indicators of compromise
 
 # Key Features
 
-* Library of known threats
-* Global sharing platform of know threats
+Identify key features of plugin.
```
Comment (jschipp-r7) [inline misp/help.md]: This got reverted

File: misp/help.md [hunk 6]
```diff
@@ -583,160 +532,216 @@ This action is used to remove a tag.
...
-* 3.0.0 - Fixed issue where Add URLs, Add Context, Add Email Sender, Add Email Subject and Add Email Recipient actions sent requests as a proposal | Fixed an issue where the distribution list was set incorrectly within Add URLs, Add Context, Create an Event, Add Email Sender, Add Email Subject, Add Email Recipient actions
-* 2.0.0 - Updated to new credential types | Update `hostname` variable in Connection to `url`
-* 1.0.0 - Add trigger. Add actions: Add Attachment, Remove Tag, Search Events, Publish. Support web server mode
-* 0.4.1 - Bug fix for CI tool incorrectly uploading plugins
-* 0.4.0 - Added add attachment feature | Update to v2 architecture
-* 0.3.0 - Add Find Event action
-* 0.2.1 - SSL bug fix in SDK
-* 0.2.0 - 8 new actions to add data to MISP: Add Email Recipient, Add Email Sender, Add Email Subject, Add Sightings, Add Tag, Add Context, Add URL, and Create An Event
-* 0.1.0 - Initial plugin
+* 1.0.0 - Initial plugin
```
Comment (jschipp-r7) [inline misp/help.md]: This got reverted

File: mxtoolbox_dns/help.md [hunk 7]
```diff
@@ -1,11 +1,17 @@
...
+Manage email server configurations and troubleshoot email delivery issues.
```
Comment (jschipp-r7) [inline mxtoolbox_dns/help.md]: This is just a list of items. Should be a long description in paragraph form.

Comment (jschipp-r7) [reply mxtoolbox_dns/help.md]: Original link is missing as well

File: viper/icon_viper/actions/list_available_compressors/schema.py [hunk 8]
```diff
@@ -56,16 +56,16 @@ class ListAvailableCompressorsOutput(komand.Output):
         "extensions": {
           "type": "array",
           "title": "Compressor Extensions",
-          "description": "Which extension are connected to archive?",
```
Comment (jschipp-r7) [inline viper/icon_viper/actions/list_available_compressors/schema.py]: We can keep these `?`s so we don't have to major version bump the plugin.

File: virustotal_yara/help.md [hunk 9]
```diff
@@ -1,13 +1,14 @@
 # Description
 
-The VirusTotal Yara plugin allows the user to analyze files with the Yara Python libarary.
-
-[VirusTotal Yara](http://virustotal.github.io/yara/) is a pattern matching swiss knife for malware researchers.
```
Comment (jschipp-r7) [inline virustotal_yara/help.md]: We should add the original link

File: virustotal_yara/plugin.spec.yaml [hunk 10]
```diff
@@ -59,7 +59,6 @@ actions:
     output:
       results:
         title: Results
-        description: 'Results returned form Yara''s analysis. Note: byte information
-          has been encoded to UTF-8'
+        description: Results returned from the Yara analysis. byte information has been encoded to UTF-8
```
Comment (jschipp-r7) [inline virustotal_yara/plugin.spec.yaml]: We'll need to fix the second sentence

File: vmray/help.md [hunk 11]
```diff
@@ -525,7 +527,7 @@ _This plugin does not contain any troubleshooting information._
 # Version History
 
 * 4.0.2 - New spec and help.md format for the Hub
-* 4.0.1 - Update actions Submit File and Submit URL to specify what analyzer to use | Update file type support for the Submit File action
+* 4.0.1 - Update actions  and Submit URL to specify what analyzer to use | Update file type support for the Submit File action
```
Comment (jschipp-r7) [inline vmray/help.md]: Revert this, original is correct

File: zenhub/komand_zenhub/actions/get_epic_data/schema.py [hunk 12]
```diff
@@ -117,7 +117,7 @@ class GetEpicDataOutput(komand.Output):
             },
             "pipeline_name": {
               "type": "string",
-              "title": "Pipelne Name",
+              "title": "Pipeline Name",
```
Comment (jschipp-r7) [inline zenhub/komand_zenhub/actions/get_epic_data/schema.py]: Requires a major version bump

File: zenhub/help.md [hunk 13]
```diff
@@ -1,14 +1,15 @@
 # Description
 
-The [Zenhub](https://www.zenhub.com) plugin allows for agile project management in GitHub. This plugin will allow you to manage issues, epics, and pipelines in a ZenHub enable Git repository.
-
+The [Zenhub](https://www.zenhub.com) plugin allows for agile project management in GitHub. This plugin will allow
+you to manage issues, epics, and pipelines in a ZenHub enabled Git repository.  Kaban boards to plan projects, reports
```
Comment (jschipp-r7) [inline zenhub/help.md]: Is your editor automatically inserting two-spaces between sentences?

File: zendesk/help.md [hunk 14]
```diff
@@ -1,6 +1,8 @@
 # Description
 
-The [Zendesk](https://www.zendesk.com) plugin helps manage communication with customers. This plugin allows you to manage tickets and users in Zendesk.
+The [Zendesk](https://www.zendesk.com) plugin helps manage communication with customers. This plugin allows you
+to manage tickets and users in Zendesk.  Customer Resource Management tool to manage tickets of user complaints
```
Comment (jschipp-r7) [inline zendesk/help.md]: Use a one-space separation between sentences

Comment (jschipp-r7) [reply zendesk/help.md]: Same with all descriptions in this PR

File: zenhub/help.md [hunk 15]
```diff
@@ -1,14 +1,15 @@
...
+to track progress and give predicted release dates.
 This plugin utilizes the [Zenhub API](https://github.com/ZenHubIO/API) to manage ZenHub Epics, Pipelines, and Estimates.
 
 # Key Features
 
 * Create an Issue
 * Create an Epic
-* Move an issue to another pipeline
+* Kaban boards
```
Comment (jschipp-r7) [inline zenhub/help.md]: Revert this change. Kaban is not so much a feature but rather what Zenhub is

File: minfraud/plugin.spec.yaml (hunk 1 above)
Comment (tylerslijboom-r7) [reply minfraud/plugin.spec.yaml]: Done

File: mxtoolbox_dns/help.md (hunk 2 above)
Comment (tylerslijboom-r7) [reply mxtoolbox_dns/help.md]: It remains, it was just moved from the start of the paragraph to the end.

File: mxtoolbox_dns/help.md (hunk 7 above)
Comment (tylerslijboom-r7) [reply mxtoolbox_dns/help.md]: Done

File: virustotal_yara/help.md (hunk 9 above)
Comment (tylerslijboom-r7) [reply virustotal_yara/help.md]: moved down the paragraph, but never removed.

File: minfraud/help.md [hunk 16]
```diff
@@ -39,8 +39,13 @@ This action can be used to retrieve intelligence information for a given payment
...
+|payment_was_authorized|boolean|None|False|Payment authorized: true/false|None|
+=======
```
Comment (jherlitz-r7) [inline minfraud/help.md]: this looks like it's got the leftovers of a git merge in it

File: minfraud/help.md [hunk 17]
```diff
@@ -161,15 +166,19 @@ This action can be used to retrieve intelligence information for all query items
 |shipping_phone_country_code|string|None|False|Country code for phone number|None|
 |order_currency|string|None|False|Currency code for the currency used|None|
 |billing_country|string|None|False|Two character country code|None|
+<<<<<<< HEAD
+|payment_was_authorized|boolean|None|False|Payment authorized: true/false|None|
+=======
 |payment_was_authorized|boolean|None|False|Payment authorized\: true/false|None|
+>>>>>>> master
```
Comment (jherlitz-r7) [inline minfraud/help.md]: remove all artifacts from git merge in minfraud help file

File: minfraud/plugin.spec.yaml [hunk 18]
```diff
@@ -6,20 +6,20 @@ title: minFraud
 vendor: rapid7
 support: community
 status: []
-description: The MaxMind minFraud services help identify fraudulent online transactions
-version: 1.0.1
+description: Help identify fraudulent online transactions, not just e-commerce; but user account creation, and application installations
```
Comment (jherlitz-r7) [inline minfraud/plugin.spec.yaml]: this description doesn't seem to make grammatical sense

File: minfraud/help.md [hunk 19]
```diff
@@ -340,6 +349,7 @@ Refer to the minFraud API documentation for explicit details.
 
 # Version History
 
+* 2.0.0 - New titles for some output variables
```
Comment (jherlitz-r7) [inline minfraud/help.md]: this change seems like it warrants a minor version bump, as titles are optional fields. They only appear in the UI and do not affect backwards compatibility. I suggest version 1.1.0, or simply leaving it as 1.0.1 because we never did a mass release of plugins after the last hackathon

File: misp/help.md [hunk 20]
```diff
@@ -1,7 +1,8 @@
 # Description
 
 [MISP](http://www.misp-project.org/) is an open source threat sharing platform.
-This plugin utilizes the [MISP API](https://circl.lu/doc/misp/automation/index.html) and leverages the [pymisp](https://github.com/MISP/PyMISP) library.
+Gather, store and then find correlations of indicators of compromise.  Quality of data is determined by the open source community.
```
Comment (jherlitz-r7) [inline misp/help.md]:
```suggestion
Gather, store and then find correlations of indicators of compromise. Quality of data is determined by the open source community.
```

File: netmiko/plugin.spec.yaml [hunk 21]
```diff
@@ -3,14 +3,15 @@ extension: plugin
 products: [insightconnect]
 name: netmiko
 title: Netmiko
-description: Netmiko is used to access network devices over SSH
-version: 1.0.1
+description: Access network devices over SSH
+version: 1.0.2
```
Comment (jherlitz-r7) [inline netmiko/plugin.spec.yaml]: I believe we want to leave the version numbers where they are right now because we never did a mass release of plugins after the last hackathon - should confirm this with @jschipp-r7

File: vmray/help.md [hunk 22]
```diff
@@ -524,8 +525,9 @@ _This plugin does not contain any troubleshooting information._
 
 # Version History
 
+* 5.0.0 - New Titles for output data (spelling corrections)
```
Comment (jherlitz-r7) [inline vmray/help.md]: keep version at 4.0.2 for now since 4.0.2 hasn't been released and title corrections only warrant a patch version bump anyway

File: vxstream_sandbox/help.md [hunk 23]
```diff
@@ -216,7 +214,8 @@ This plugin does not contain any troubleshooting information.
 
 # Version History
 
-* 3.0.0 - New spec and help.md format for the Hub | Fix spelling of variable titled Additional Context
+* 3.0.0 - Updated variable titles, spelling mistakes
```
Comment (jherlitz-r7) [inline vxstream_sandbox/help.md]: looks like this should stay as 2.0.1

File: wazuh_ossec/help.md [hunk 24]
```diff
@@ -603,7 +607,8 @@ This plugin does not contain any troubleshooting information.
 
 # Version History
 
-* 1.0.1 - New spec and help.md format for the Hub | Spelling fixes in descriptions
+* 2.0.0 - Removed colon from variable names wazuh-modules_oscap and wazuh-modules_database
```
Comment (jherlitz-r7) [inline wazuh_ossec/help.md]: same as others, revert 1.0.1

File: wazuh_ossec/plugin.spec.yaml [hunk 25]
```diff
@@ -3,9 +3,9 @@ extension: plugin
 products: [insightconnect]
 name: wazuh_ossec
 title: Wazuh OSSEC
-description: Provides security visibility, compliance and infrastructure monitoring
+description: Open source security platform, provides security visibility, compliance and infrastructure monitoring
```
Comment (jherlitz-r7) [inline wazuh_ossec/plugin.spec.yaml]:
```suggestion
description: Open source security platform providing security visibility, compliance, and infrastructure monitoring
```

File: whois/bin/komand_whois [hunk 26]
```diff
@@ -7,7 +7,7 @@ from komand_whois import connection, actions, triggers
 Name = 'WHOIS'
 Vendor = 'rapid7'
 Version = '1.0.6'
-Description = 'The WHOIS plugin enables address and domain lookups in the WHOIS databases'
+Description = 'The WHOIS plugin enables address and domain look ups in the WHOIS databases'
```
Comment (jherlitz-r7) [inline whois/bin/komand_whois]:
```suggestion
Description = 'The WHOIS plugin enables address and domain lookups in the WHOIS databases'
```
pretty sure a "lookup" is one word

File: wordpress/help.md [hunk 27]
```diff
@@ -1,10 +1,11 @@
 # Description
 
-The [WordPress](https://wordpress.com/) plugin allows you to manage users on the WordPress platform
+The [WordPress](https://wordpress.com/) plugin allows you to manage users on the WordPress platform.  Only users can be deleted or suspended from the wordpress application.
```
Comment (jherlitz-r7) [inline wordpress/help.md]:
```suggestion
The [WordPress](https://wordpress.com/) plugin allows you to manage users on the WordPress platform. Only users can be deleted or suspended from the wordpress application.
```

File: wordpress/plugin.spec.yaml [hunk 28]
```diff
@@ -3,7 +3,7 @@ extension: plugin
 products: [insightconnect]
 name: wordpress
 title: WordPress
-description: The WordPress plugin allows for user management on the WordPress platform
+description: User management on the WordPress platform only, suspend or delete users
```
Comment (jherlitz-r7) [inline wordpress/plugin.spec.yaml]:
```suggestion
description: Suspend and delete users from a WordPress instance
```

File: zendesk/bin/komand_zendesk [hunk 29]
```diff
@@ -7,7 +7,7 @@ from komand_zendesk import connection, actions, triggers
 Name = 'Zendesk'
 Vendor = 'rapid7'
 Version = '1.1.1'
-Description = 'The Zendesk plugin helps manage communication with customers. This plugin allows you to manage tickets and users in Zendesk of data'
+Description = 'Manage customer communications. Manage tickets and users in Zendesk data'
```
Comment (jherlitz-r7) [inline zendesk/bin/komand_zendesk]:
```suggestion
Description = 'Regulate customer communications by managing tickets and users in Zendesk data'
```

File: zendesk/help.md [hunk 30]
```diff
@@ -1,6 +1,6 @@
 # Description
 
-The [Zendesk](https://www.zendesk.com) plugin helps manage communication with customers. This plugin allows you to manage tickets and users in Zendesk.
+The [Zendesk](https://www.zendesk.com) plugin helps manage communication with customers. This plugin allows you to manage tickets and users in Zendesk.  Customer Resource Management tool to manage tickets of user complaints and support issues.
```
Comment (jherlitz-r7) [inline zendesk/help.md]:
```suggestion
The [Zendesk](https://www.zendesk.com) plugin helps manage communication with customers. This plugin allows you to manage tickets and users in Zendesk. Customer Resource Management tool to manage tickets of user complaints and support issues.
```

File: zendesk/bin/komand_zendesk (hunk 29 above)
Comment (jherlitz-r7) [reply zendesk/bin/komand_zendesk]: this should be altered in the spec and in the help and then run regenerate to fix it in this bin file

File: zenhub/help.md [hunk 31]
```diff
@@ -1,14 +1,12 @@
 # Description
 
-The [Zenhub](https://www.zenhub.com) plugin allows for agile project management in GitHub. This plugin will allow you to manage issues, epics, and pipelines in a ZenHub enable Git repository.
-
+The [Zenhub](https://www.zenhub.com) plugin allows for agile project management in GitHub. This plugin will allow you to manage issues, epics, and pipelines in a ZenHub enabled Git repository.  Kaban boards to plan projects, reports to track progress and give predicted release dates.
```
Comment (jherlitz-r7) [inline zenhub/help.md]: the last sentence in this description doesn't really make sense - it's more of a sentence fragment

File: zenhub/plugin.spec.yaml [hunk 32]
```diff
@@ -3,7 +3,7 @@ extension: plugin
 products: [insightconnect]
 name: zenhub
 title: ZenHub
-description: The ZenHub plugin allows for agile project management in GitHub
+description: Agile project management in GitHub, project road maps, kaban boards all attached to repositories
```
Comment (jherlitz-r7) [inline zenhub/plugin.spec.yaml]:
```suggestion
description: Agile project management in GitHub, project road maps, and kanban boards all attached to repositories
```

Comment (jherlitz-r7) [reply zenhub/plugin.spec.yaml]: still might want to revise this more to make it clear exactly what this plugin is for

File: minfraud/help.md (hunk 17 above)
Comment (tylerslijboom-r7) [reply minfraud/help.md]: That face looks more disappointed to me, than confused.  This is sloppy on my part, and I apologize.

File: minfraud/help.md (hunk 19 above)
Comment (tylerslijboom-r7) [reply minfraud/help.md]: title changes are major version updates.

File: netmiko/plugin.spec.yaml (hunk 21 above)
Comment (tylerslijboom-r7) [reply netmiko/plugin.spec.yaml]: "Yeah, it’s a major version bump if titles change"

File: wordpress/help.md (hunk 27 above)
Comment (tylerslijboom-r7) [reply wordpress/help.md]: Double spaces after a period is keyboard muscle memory for me.  removed nonetheless

File: minfraud/help.md (hunk 19 above)
Comment (jschipp-r7) [reply minfraud/help.md]: Latest minFraud is https://market.komand.com/plugins/rapid7/minfraud/1.0.0

so we should combine entries, and remove the 1.0.1 listing from the Changelog since it wasn't released

```suggestion
* 2.0.0 - New spec and help.md format for the Hub | New titles for some output variables
```

File: misp/plugin.spec.yaml [hunk 33]
```diff
@@ -1198,7 +1198,7 @@ actions:
         type: string
         required: false
         enum:
-        - Don't search on
+        - Do not search on
```
Comment (jschipp-r7) [inline misp/plugin.spec.yaml]: Need a major version bump for these too since the values change. Any workflow depending on the previous value will not work if the plugin is upgraded.

File: netmiko/help.md [hunk 34]
```diff
@@ -98,6 +101,7 @@ _This plugin does not contain any troubleshooting information._
 
 # Version History
 
+* 1.0.2 - Updated python libraries
```
Comment (jschipp-r7) [inline netmiko/help.md]: 1.0.1 hasn't been released https://market.komand.com/plugins/rapid7/netmiko/1.0.0

So set version to 1.0.1, regenerate, and combine the changelog lines.

File: newrelic/plugin.spec.yaml [hunk 35]
```diff
@@ -49,7 +49,7 @@ types:
       type: string
 actions:
   list_users:
-    title: List User
+    title: List Users
```
Comment (jschipp-r7) [inline newrelic/plugin.spec.yaml]: Major version bump

File: wazuh_ossec/help.md [hunk 36]
```diff
@@ -603,7 +607,8 @@ This plugin does not contain any troubleshooting information.
...
+* 1.0.1 - New spec and help.md format for the Hub
```
Comment (jschipp-r7) [inline wazuh_ossec/help.md]: Latest one is https://market.komand.com/plugins/rapid7/wazuh_ossec/1.0.0 so 1.0.1 was never released.

Combine this line with 2.0.0 and release the major version of 2.0.0

File: zenhub/plugin.spec.yaml [hunk 37]
```diff
@@ -75,7 +75,7 @@ types:
       description: Position
       required: false
     pipeline_name:
-      title: Pipelne Name
+      title: Pipeline Name
```
Comment (jschipp-r7) [inline zenhub/plugin.spec.yaml]: Major version bump and regenerate

File: zenhub/plugin.spec.yaml [hunk 38]
```diff
@@ -3,7 +3,7 @@ extension: plugin
 products: [insightconnect]
 name: zenhub
 title: ZenHub
-description: The ZenHub plugin allows for agile project management in GitHub
+description: Agile project management in GitHub.   Manage issues and epics
```
Comment (jschipp-r7) [inline zenhub/plugin.spec.yaml]:
```suggestion
description: Manage issues and epics in Zenhub, an agile project management tool for GitHub
```

File: newrelic/plugin.spec.yaml (hunk 35 above)
Comment (tylerslijboom-r7) [reply newrelic/plugin.spec.yaml]: I reverted back, I just want to avoid version bumps unless I have to.
